src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/executor_pa.cpp

```diff
@@ -1416,7 +1465,23 @@ struct MHA {
             }
         });
 
-        parallel_for2d_dynamic(attn_work_count, Hk, [&](size_t w, size_t hk) {
+        // loop along HK dimension: if mixed first/second token and elements count is enough, loop HK to reuse KV in the CPU cache
+        //    else if elements count is small, prefer to loop H to get more work to avoid thread imbalance
+        bool loop_hk = _workitems.get_reorder_max_batch_size() == past_lens.m_dims[0] ||        // if only first token, loop H
+            attn_work_count * Hk <= 2 * _helper._nthr ? false : true;                           // or less than 2 work items per thread, loop H
```

Comment:
Please align the threshold between `attn_work_count * Hk <= 2 * _helper._nthr` in `exec_loop_mixed` and `bool loop_hk = B * kv_len_in_blocks * _Hk > 2 * _nthr` in `exec_loop_bhl`

Reply:
Please also apply the threshold comment to corresponding place in `exec_loop_bhl`

Reply:
Done.